cmake/Modules/FindTriSYCL.cmake

```diff
@@ -0,0 +1,93 @@
+#.rst:
+# FindTriSYCL
+#---------------
+#
+#   TODO : add Copyright
+
+#########################
+#  FindTriSYCL.cmake  
+#########################
+#
+#  Tools for finding and building with triSYCL.
+#
+#  User must define:
+#	TRISYCL_PACKAGE_ROOT_DIR pointing to triSYCL
+#		available : https://github.com/keryell/triSYCL
+#	COMPUTE_PACKAGE_ROOT_DIR pointing to Compute
+#		available : https://github.com/boostorg/compute
+#
+
+# Require CMake version 3.2.2 or higher
+cmake_minimum_required(VERSION 3.2.2)
+
+# Check that a supported host compiler can be found
+if(CMAKE_COMPILER_IS_GNUCXX)
+    # Require at least gcc 4.8
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8)
+      message(FATAL_ERROR
+        "host compiler - Not found! (gcc version must be at least 4.8)")
+    # Require the GCC dual ABI to be disabled for 5.1 or higher
+    elseif (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.1)
+      set(COMPUTECPP_DISABLE_GCC_DUAL_ABI "True")
+      message(STATUS
+        "host compiler - gcc ${CMAKE_CXX_COMPILER_VERSION} (note pre 5.1 gcc ABI enabled)")
+    else()
+      message(STATUS "host compiler - gcc ${CMAKE_CXX_COMPILER_VERSION}")
+    endif()
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    # Require at least clang 3.6
+    if (${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS 3.6)
+      message(FATAL_ERROR
+        "host compiler - Not found! (clang version must be at least 3.6)")
+    else()
+      message(STATUS "host compiler - clang ${CMAKE_CXX_COMPILER_VERSION}")
+    endif()
+else()
+  message(WARNING
+	  "host compiler - Not found! (triSYCL supports GCC and Clang, see readme)")
+endif()
+
+set(COMPUTECPP_64_BIT_DEFAULT ON)
+option(COMPUTECPP_64_BIT_CODE "Compile device code in 64 bit mode"
+        ${COMPUTECPP_64_BIT_DEFAULT})
+mark_as_advanced(COMPUTECPP_64_BIT_CODE)
+
+# Find OpenCL package
+find_package(OpenCL REQUIRED)
+
+# Find triSYCL packagee
+if(NOT TRISYCL_PACKAGE_ROOT_DIR)
+	message(FATAL_ERROR "triSYCL package - Not found! (please set TRISYCL_PACKAGE_ROOT_DIR): ${TRISYCL_PACKAGE_ROOT_DIR}")
+else()
+	message(STATUS "triSYCL package - Found")
+endif()
+option(TRISYCL_PACKAGE_ROOT_DIR "Path to the triSYCL Package")
+
+# Find Compute packagee
```

Comment:
It is not clear to me how it can works with pre-installed normal Boost.Compute.
Just rely on the normal CMake way to add directories to search for some include files?